lib/kernel/test/pg_SUITE.erl

```diff
@@ -67,13 +67,46 @@ suite() ->
     [{timetrap, {seconds, 60}}].
 
 init_per_testcase(TestCase, Config) ->
-    {ok, _Pid} = pg:start_link(TestCase),
-    Config.
+    {ok, Pid} = pg:start_link(TestCase),
+    trace_start(TestCase, Config, Pid).
 
-end_per_testcase(TestCase, _Config) ->
+end_per_testcase(TestCase, Config) ->
     gen_server:stop(TestCase),
+    trace_end(Config),
     ok.
 
+trace_start(forced_sync, Config, Tracee) ->
+    Tracer = spawn_link(fun() -> tracer() end),
+    1 = erlang:trace(Tracee, true, ['receive', {tracer, Tracer}, timestamp]),
+    [{tracer, Tracer} | Config];
+trace_start(_, Config, _) ->
+    Config.
+
+trace_end(Config) ->
+    case proplists:get_value(tracer, Config) of
+        undefined -> ok;
+        Tracer ->
+            Mon = erlang:monitor(process, Tracer),
+            Tracer ! flush,
+            normal = receive
+                         {'DOWN', Mon, process, Tracer, R} -> R
+                     end
+    end.
+
+tracer() ->
+    receive flush -> ok end,
+    io:format("Flush trace messages:\n"),
```

Comment:
Do you mind removing these debug messages in the final version?

Reply:
Yes, that was the plan. Just want to make sure pg_SUITE runs clean for a while before I merge.
Thanks for the reminder.

Reply:
I might keep the trace code deactivated, if that's ok with you. I have already rewritten it twice.

Reply:
That'd be fine, but then could you move it next to `sync`/`sync_via` functions to preserve the overall structure of the suite?